### PR TITLE
Fixing choropleth calculation bug

### PR DESCRIPTION
**Pull request by emre2038**
## Description
Searching for the reason and fixing the issue that causes choropleth values bigger than 100%

## Related Issue
https://github.com/OpenUpSA/wazimap-ng-ui/issues/246

## How to test it locally
* set `hostname` parameter in `index.js` as 'sifar-wazi.openup.org.za'
* navigate to `/#geo:199014`
* open data mapper
* select Demographics -> Population -> Population by age -> 00 - 04
* confirm that the choropleth values are correct

## Screenshots
image

## Changelog
* removed the unnecessary if check in `subindicator_calculator.js`. we already assume each value is `0` if they are `undefined`. that additional if check breaks the calculation 

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design matches the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

**Comment by milafrerichs** [comment]
@emre2038 I added a test for it and refactored the code based on the test. 
these kind of small bugs are a great way to introduce unit tests. it was easy enough to create a small test for it and then refactor the code a bit more.
the naming was a bit confusing so I tried my best to come up with better names.

File: src/js/map/choropleth/subindicator_calculator.js [hunk 1]
```diff
@@ -2,25 +2,19 @@ import {percFmt, numFmt} from '../../utils'
...
+        if (typeof args.data === 'undefined') {
             return null;
         }
 
-        const result = Object.entries(args.data).map(childGeography => {
-            const code = childGeography[0];
-            const count = childGeography[1];
-            const universe = args.subindicatorArr.reduce((el1, el2) => {
-                if (el2.children != undefined && el2.children[code] != undefined) {
-                    let e1 = typeof el1 === 'undefined' ? 0 : el1;
```
Comment (milafrerichs) [inline src/js/map/choropleth/subindicator_calculator.js:14]: this check was unnecessary because e1 is the total and can never be `undefined`. it starts with `0` the value we call `reduce` with and then is always the return value from the method itself.

File: src/js/map/choropleth/subindicator_calculator.js [hunk 2]
```diff
@@ -2,25 +2,19 @@ import {percFmt, numFmt} from '../../utils'
 
 export default class SubindicatorCalculator {
     calculate(args) {
-        if (typeof args.data === 'undefined'){
+        if (typeof args.data === 'undefined') {
             return null;
         }
 
-        const result = Object.entries(args.data).map(childGeography => {
-            const code = childGeography[0];
```
Comment (milafrerichs) [inline src/js/map/choropleth/subindicator_calculator.js:10]: I refactored this to be in the method call of map.

File: src/js/map/choropleth/subindicator_calculator.js [hunk 3]
```diff
@@ -2,25 +2,19 @@ import {percFmt, numFmt} from '../../utils'
...
-                }
+        const result = Object.entries(args.data).map(([code, codeValue]) => {
+            const sumSubindicatorsValueForCode = args.subindicatorArr.reduce((sum, subindicator) => {
+                let childrenValue = subindicator.children[code] || 0;
+                return sum + childrenValue;
             }, 0);
 
-            const val = count / universe;
-            return {code: code, val: val};
+            const percentage = codeValue / sumSubindicatorsValueForCode;
```
Comment (milafrerichs) [inline src/js/map/choropleth/subindicator_calculator.js:15]: I renamed `val` to `percentage` because that's what it us. and `universe` to be `sumSubindicatorsValueForCode` to make it more verbose.

File: src/js/map/choropleth/subindicator_calculator.js [hunk 4]
```diff
@@ -2,25 +2,19 @@ import {percFmt, numFmt} from '../../utils'
...
-                if (el2.children != undefined && el2.children[code] != undefined) {
-                    let e1 = typeof el1 === 'undefined' ? 0 : el1;
-                    let e2 = typeof  el2.children[code] === 'undefined' ? 0 :  el2.children[code];
-                    let total = e1 + e2;
-
-                    return total;
-                }
+        const result = Object.entries(args.data).map(([code, codeValue]) => {
+            const sumSubindicatorsValueForCode = args.subindicatorArr.reduce((sum, subindicator) => {
+                let childrenValue = subindicator.children[code] || 0;
```
Comment (milafrerichs) [inline src/js/map/choropleth/subindicator_calculator.js:11]: I used `||` here because that's how to best check for `undefined` and assign a "default" value
